autotest/test_budget.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import pytest
+
+from pynhm.base.budget import Budget
+
+
+def test_budget():
```

Comment:
is this init awkward?

Reply:
As far as init goes, I think a main point is that the list of input/output/ds terms are defined up front.  And we've found that they should all be defined up front.  We do not allow another term to be inserted later in the simulation.  I think what you have done requires that.  Your init method requires that initial rates are specified for each variable, but I'm not sure we need to do that.  If anything we could allow specification of initial volumes, but I think that would be different.  We also have a reset method, which sets all the terms to zero, and we call that before we start adding new values.  Bottom line is that it makes sense to me that init requires a dictionary of input/output/ds.

Reply:
rates dont have to be specified on init now.
budgets are also somewhat protective of changing terms (probably not comprehensive).